sql/src/main/java/org/apache/druid/sql/calcite/schema/DruidSchemaManager.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.calcite.schema;
+
+import org.apache.druid.sql.calcite.table.DruidTable;
+
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * This interface provides a map of datasource names to {@link DruidTable} objects, used by the {@link DruidSchema}
+ * class as the SQL planner's view of Druid datasource schemas. If a non-default implementation is provided,
+ * the segment metadata polling-based view of the Druid tables will not be built in DruidSchema.
+ */
+public interface DruidSchemaManager
```

Comment:
shall this be marked as @ExtensionPoint and also as @UnstableAPI

Reply:
Added the annotations